### PR TITLE
[[ Bug 16584 ]] Ensure source and pattern marks are in sync when wildcard matching

### PR DESCRIPTION
**Pull request by livecodeali**

### CODE REVIEW

File: tests/lcs/core/strings/filter.livecodescript [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+/*
+	Filter wildcard special characters:
+	* - matches any number of occurrences of any character
+	? - matches a single occurrence of any character
+	[abc] - matches any of a, b or c
+	[a-m] - matches any lowercase character in the first half of the alphabet
+	[!<string>] - matches iff [<string>] does not match
```
Comment (runrevmark) [inline tests/lcs/core/strings/filter.livecodescript]: In this correct? I thought [!...] was negated character class so [!abc] matches any character except for a, b or c.

Comment (livecodeali) [inline tests/lcs/core/strings/filter.livecodescript]: Yes, you're right, thanks :-) Which reminds me, that needs to be added to the dictionary!

Comment (runrevmark) [inline tests/lcs/core/strings/filter.livecodescript]: There was a 'regex reference card' somewhere in the IDE at one point... We should probably try and find that and put into the dictionary / guides in a more appropriate place!

**Comment by runrevmark** [comment]
@livecode-vulcan review ok ad57929

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok ad579293fe2e77a0cda0ce54c59cff6c272e9618

**Comment by livecode-vulcan** [comment]
😎 test success ad579293fe2e77a0cda0ce54c59cff6c272e9618
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/1453)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/1473)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/1449)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/1460)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/1472)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/1473)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/1475)

**Pull request merged**